eth/stagedsync/stage_execute.go

```diff
@@ -534,7 +532,17 @@ func SpawnExecuteBlocksStage(s *sync_stages.StageState, u sync_stages.Unwinder,
 		to = noProgressTo
 	}
 
+	// limit execution to 100 blocks at a time for faster sync near tip
+	// [TODO] remove it after Interhashes  incremental is optimized
 	total := to - stageProgress
+	if total > 100 && total < 100000 {
```

Comment:
link this to the command-line flag for incremental interhashes: `zkevm.rebuild-tree-after` + make this flag `100` by default.